mlir/lib/Dialect/Affine/IR/AffineOps.cpp

```diff
@@ -427,6 +427,25 @@ bool mlir::affine::isValidSymbol(Value value) {
   return false;
 }
 
+/// A utility function to check if a value is defined at the top level of
+/// `region` or is an argument of `region` or dominates the region.
```

Comment:
Where is the "dominate" aspect checked?
As far as I can tell you're only checking if "a value is defined in the current region or implicitly captured".

Isn't `areValuesDefinedAbove()` suitable here?

Reply:
> the underlying code checks the is isolated from above constraint, areValuesDefinedAbove isn't equivalent

Seems like `areValuesDefinedAbove` should be updated instead: I don't think we should have the same utility with a different implementation multiple-times in the codebase.

That said: checking for traits may very well be more expensive than getting all the parent regions.